### PR TITLE
Heroku Button

### PR DESCRIPTION
**Pull request by NumberC**
### Describe the changes you have made in this pr -
Added a Heroku button in the ReadMe.md file. If you try clicking it in my local repo, it won't work because it is meant to work after my change gets merged. This is because it takes the URL of the GitHub repo to work. Obviously, the URL for my repo will be different from the actual repo.

### Screenshots of the changes (If any) -
When the user clicks the button, this should show (Note: instead of NumberC/CircuitVerse/#HerokuBtn, it will say something like CircuitVerse/CircuitVerse/#master when it gets merged):
Part1

### CODE REVIEW

**Comment by NumberC** [comment]
@tachyons To get rid of the unrelated changes in the PR, do I have to go through each commit and do "git rever <SSH>"? Or is there another way?

**Comment by tachyons** [comment]
Delete the forked repo and fork a fresh repository and make changes

**Comment by NumberC** [comment]
@tachyons Won't that get rid of all my work for every other PR?

**Comment by Ananthavijay** [comment]
- @NumberC Could have check this PR again. When I try to deploy my fork with the changes you have made, the deploy fails.
- Log : 
   Running: rake assets:precompile
       Yarn executable was not detected in the system.
       Download Yarn at https://yarnpkg.com/en/docs/install
       rake aborted!
       SassC::SyntaxError: Error: File to import not found or unreadable: bootstrap/scss/bootstrap.scss.
- For this kind of tasks, First, make a test deploy and then make a PR if the test deploy is successful

**Comment by NumberC** [comment]
@Ananthavijay This is a common issue that isn't related to my changes. This issue is addressed in the Readme in "Additional setup instructions"

**Comment by NumberC** [comment]
@tachyons Finally! I think I got it now! I compared my fork and undid all the changes in files, commited that, added the relevant changes, and commited those. It should be fine now.

**Comment by tachyons** [comment]
Deploy to heorku button should deploy app to heroku, that includes adding buildpacks and dependencies , otherwise you will get error as Vijay mentioned

**Comment by NumberC** [comment]
@tachyons Oh! I misunderstood that message. I believe the reason it may not be working as you expect is because I am using the CircuitVerse URL, instead of my own. When you merge it, it should work. For instance, I will change it now to my URL now so it should work.

**Comment by NumberC** [comment]
Nevermind, it still failed to deploy. Sorry

**Comment by tachyons** [comment]
Can you url of provide sample app deployed with your configuration ?

**Comment by NumberC** [comment]
I am not sure I understand the question. Heroku just tries to deploy the HerokuBtn fork from my repo, but there's an error related to Yarn and I believe my installation of Yarn. There is a link to CircuitVerse.org in the app.json file.

**Comment by tachyons** [comment]
Have you tried deploy button in the screenshot you provided? Did it work, If then provide the url of that installation

File: config/environments/production.rb [hunk 1]
```diff
@@ -97,8 +97,8 @@
      :enable_starttls_auto => true,
  }
 
-  config.vapid_public_key = ENV["VAPID_PUBLIC_KEY"]
-  config.vapid_private_key = ENV["VAPID_PRIVATE_KEY"]
+  config.vapid_public_key = ENV["VAPID_PUBLIC_KEY"] || "BGxnigbQCa435vZ8_3uFdqLC0XJHXtONgEdI-ydMMs0JaBsnpUfLxR1UDagq6_cDwHyhqjw77tTlp0ULZkx8Xos="
```
Comment (tachyons) [inline config/environments/production.rb]: Refer this https://devcenter.heroku.com/articles/app-json-schema#env

**Comment by alanjsph** [comment]
Code looks good. Consider moving deployment specific environment variables to app.json as Aboobacker has mentioned.

Also do not put commit messages like "Hope this works" :) . Keep the messages in such a way that what you have done in that particular PR, so it makes the life of other people looking at your code base easier.

**Comment by NumberC** [comment]
@alanjsph I edited my commit messages! Sorry, I couldn't change the one that you pointed out. It was too far back in the history

**Comment by satu0king** [comment]
@Ananthavijay does this help you?

**Comment by Ananthavijay** [comment]
@satu0king first the repo should properly work when deployed in heroku, after then the heroku button can be implemented.

**Pull request closed without merging**